go/enclave/db/storage.go

```diff
@@ -265,9 +265,15 @@ func (s *storageImpl) writeLog(l *types.Log, stateDB *state.StateDB, dbBatch *sq
 		isLifecycle = isLifecycle && !isUserAccount
 	}
 
+	// normalise the data field to nil to avoid duplicates
+	data := l.Data
+	if len(data) == 0 {
+		data = nil
+	}
```

Comment:
Out of the loop on the context, but if `l.Data := []byte{0, 0, 0}` then the len is not nil and it will execute in the sql below. Is that ok in this scenario ?

Reply:
that's ok. 
what was happening is that sometimes the data field was nil, sometimes it was empty for the same log entry